engine/api/workflow_run.go

```diff
@@ -787,184 +788,175 @@ func (api *API) postWorkflowRunHandler() service.Handler {
-					return sdk.WrapError(err, "unable to start transaction")
-				}
-				if err := workflow.SyncAsCodeEvent(ctx, tx, api.Cache, p, wf); err != nil {
-					tx.Rollback() // nolint
-					return err
-				}
-				if err := tx.Commit(); err != nil {
-					return sdk.WrapError(err, "unable to commit transaction")
-				}
+			})
+			if errWf != nil {
+				return sdk.WrapError(errWf, "unable to load workflow %s", name)
 			}
 
+			// Check allow as code
 			if wf.FromRepository != "" {
+				enabled, has := p.Features[feature.FeatWorkflowAsCode]
 				if has && !enabled {
-					return sdk.WrapError(sdk.ErrForbidden, "postWorkflowRunHandler> %s not allowed for project %s", feature.FeatWorkflowAsCode, p.Key)
-				}
-				proj, errp := project.Load(api.mustDB(), api.Cache, key, u,
-					project.LoadOptions.WithGroups,
-					project.LoadOptions.WithApplicationVariables,
-					project.LoadOptions.WithApplicationWithDeploymentStrategies,
-					project.LoadOptions.WithEnvironments,
-					project.LoadOptions.WithPipelines,
-					project.LoadOptions.WithClearKeys,
-					project.LoadOptions.WithClearIntegrations,
-				)
-
-				if errp != nil {
-					return sdk.WrapError(errp, "postWorkflowRunHandler> Cannot load project %s", key)
-				}
-				// Get workflow from repository
-				var errCreate error
-				asCodeInfosMsg, errCreate = workflow.CreateFromRepository(ctx, api.mustDB(), api.Cache, proj, wf, *opts, u, project.DecryptWithBuiltinKey)
-				if errCreate != nil {
-					var msgListString string
-					if len(asCodeInfosMsg) > 0 {
-						msgListString = strings.Join(translate(r, asCodeInfosMsg), " ")
-					}
-					return sdk.WrapError(errCreate, "postWorkflowRunHandler> Unable to get workflow from repository.%s", msgListString)
-				}
-			} else {
-				var errl error
-				options := workflow.LoadOptions{
-					DeepPipeline: true,
-					Base64Keys:   true,
-				}
-				wf, errl = workflow.Load(ctx, api.mustDB(), api.Cache, p, name, u, options)
-				if errl != nil {
-					return sdk.WrapError(errl, "postWorkflowRunHandler> Unable to load workflow %s/%s", key, name)
+					return sdk.WrapError(sdk.ErrForbidden, "as code is not allowed for project %s", p.Key)
 				}
 			}
-		}
-		if name != wf.Name {
-			return sdk.WrapError(sdk.ErrWorkflowInvalid, "workflow %s asked, but workflow %s found", name, wf.Name)
-		}
 
-		report, errS := startWorkflowRun(ctx, api.mustDB(), api.Cache, p, wf, lastRun, opts, u, asCodeInfosMsg)
+			// Check node permission
+			if !permission.AccessToWorkflowNode(wf, &wf.WorkflowData.Node, u, permission.PermissionReadExecute) {
+				return sdk.WrapError(sdk.ErrNoPermExecution, "not enough right on node %s", wf.WorkflowData.Node.Name)
+			}
 
-		if errS != nil {
-			return sdk.WrapError(errS, "postWorkflowRunHandler> Unable to start workflow %s/%s", key, name)
+			// CREATE WORKFLOW RUN
+			var errCreateRun error
+			lastRun, errCreateRun = workflow.CreateRun(api.mustDB(), wf, opts, u)
+			if errCreateRun != nil {
+				return errCreateRun
```

Comment:
```suggestion
				return sdk.WithStack(errCreateRun)
```

Reply:
no need because err from CreateRun is already wrapper